src/serializers/ob_type.rs

```diff
@@ -333,3 +411,47 @@ pub enum ObType {
     // unknown type
     Unknown,
 }
+
+impl PartialEq for ObType {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
```

Comment:
It feels easy for this to break when adding new `ObType` variants, how about having a fast-path for equal values first which also deals with unknown?

```suggestion
        if (self as u8) == (other as u8) {
            !matches(self, Self::Unknown)
        } else match (self, other) {
```

... you can then remove all the `(X, X) => true` branches below.